lib/components/CharacterCount.js

```diff
@@ -0,0 +1,27 @@
+import React, { Component } from 'react';
+import firebase, { reference } from '../firebase';
+import MessageInput from './MessageInput';
+
+
+export default class CharacterCount extends Component {
+  constructor(newMessage) {
+    super();
+    this.state = {
+      charactersLeft: 140
+    };
+  }
+
+  componentWillReceiveProps(nextProps) {
```

Comment:
It was something Brittany suggested to me for number guesser to help when a component depends on the changing state of another component.  I think nextProps is just a naming convention like "handleClick"  here is an excerpt from the docs ...

Use this as an opportunity to react to a prop transition before render() is called by updating the state using this.setState(). The old props can be accessed via this.props. Calling this.setState() within this function will not trigger an additional render.

componentWillReceiveProps: function(nextProps) {
  this.setState({
    likesIncreasing: nextProps.likeCount > this.props.likeCount
  });
}